app/models/data.js

```diff
@@ -52,4 +52,26 @@ Data.prototype.delete = function (key, callback) {
   });
 };
 
+Data.prototype.deleteOlderThanDate = function (date, callback) {
+  this.db.find({}, (err, docs) => {
+    if (err) {
+      return callback(err);
+    }
+
+    if (docs.length > 0) {
+      docs.forEach((item)=>{
+        let updatedAtDate = new Date(item.updatedAt);
+        if (updatedAtDate.getTime() < date) {
```

Comment:
A db.find() is run with no parameters, then a bunch of work is done to filter based on date.
Can't the db handle some kind of query?